### PR TITLE
Ignore fetch/push options in configuration of remote repos

### LINKED ISSUE #11424: [NBug] Unable to update remote pushurl for command output: origin	...
### Environment

- Git Extensions 4.2.1.17611
- Build b0c0b284876b63c52c77195ada82a85bfadbb083
- Git 2.43.0.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.25
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

```
System.Exception: Unable to update remote pushurl for command output: origin	git@github.com:flannelhead/jsmn-stream.git (push)
   at Task<IReadOnlyList<Remote>> GitCommands.GitModule.GetRemotesAsync()+ParseRemotes(ExecutionResult result) in C:/projects/gitextensions/GitCommands/Git/GitModule.cs:line 2371
   at async Task<IReadOnlyList<Remote>> GitCommands.GitModule.GetRemotesAsync() in C:/projects/gitextensions/GitCommands/Git/GitModule.cs:line 2336
   at void Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTask<T>.CompleteOnCurrentThread()
   at Nodes GitUI.LeftPanel.RemoteBranchTree.FillTree(IReadOnlyList<IGitRef> branches, CancellationToken token) in C:/projects/gitextensions/GitUI/LeftPanel/RemoteBranchTree.cs:line 29
   at async Task<Nodes> GitUI.LeftPanel.BaseRefTree.LoadNodesAsync(CancellationToken token, Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs) in C:/projects/gitextensions/GitUI/LeftPanel/BaseRefTree.cs:line 40
   at async JoinableTask GitUI.LeftPanel.Tree.ReloadNodesDetached(Func<CancellationToken, Func<RefsFilter, IReadOnlyList<IGitRef>>, Task<Nodes>> loadNodesTask, Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs)+(?) => { } in C:/projects/gitextensions/GitUI/LeftPanel/Tree.cs:line 98
   at async Task Microsoft.VisualStudio.Threading.ThreadingTools.WithCancellationSlow(Task task, bool continueOnCapturedContext, CancellationToken cancellationToken)
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 28
```

### Steps to reproduce

Create minimal clone with
`git clone --filter=blob:none git@github.com:flannelhead/jsmn-stream.git`
 and try to open GitExtensions on this directory

### Did this work in previous version of GitExtensions?

GitExtensions 4.2 works without exception.

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by mstv**
Fixes #11424

## Proposed changes

- Ignore fetch/push options in `GetRemotesAsync`

## Screenshots 

N/A

## Test methodology 

- extended testcase

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/Git/GitModule.cs [hunk 1]
```diff
@@ -2327,7 +2327,7 @@ public IReadOnlyList<string> GetRemoteNames()
                 .ToList();
         }
 
-        private static readonly Regex _remoteVerboseLineRegex = new(@"^(?<name>[^	]+)\t(?<url>.+?) \((?<direction>fetch|push)\)$", RegexOptions.Compiled);
+        private static readonly Regex _remoteVerboseLineRegex = new(@"^(?<name>[^	]+)\t(?<url>.+?) \((?<direction>fetch|push)\)(?<option>\s*\[[^\]]*])?$", RegexOptions.Compiled);
```
Comment (gerhardol) [inline GitCommands/Git/GitModule.cs:2330]: Capturing is affecting performance.
More important, I would look for how that data is used.
```suggestion
       // Ignore trailing options
        private static readonly Regex _remoteVerboseLineRegex = new(@"^(?<name>[^	]+)\t(?<url>.+?) \((?<direction>fetch|push)\)(?:\s*\[[^\]]*])?$", RegexOptions.Compiled | RegexOptions.ExplicitCapture);
```

**Review by gerhardol — COMMENTED** [review]
+1 due to comments, not opposing merging

File: GitCommands/Git/GitModule.cs (hunk 1 above)
Comment (mstv) [reply GitCommands/Git/GitModule.cs]: > I would look for how that data is used

I guess git uses them automatically.

Comment (gerhardol) [reply GitCommands/Git/GitModule.cs]: > I guess git uses the option(s) automatically.

I just meant `(?<option`
ExplicitCapture is not set, OK

**Pull request merged**